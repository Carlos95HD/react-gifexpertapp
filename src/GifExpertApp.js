import React from 'react';
import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

  // const categories = [ 'One Punch', 'Samurai X', 'Dragon Ball' ];
  const [categories, setCategories] = useState([ 'Welcome' ])

  // const handleAdd = () => {
  //   setCategories([...categories, 'Simpsons'])
  // }

  return (
    <>
    <h2 className='text-4xl font-mono text-gray-800 animate__animated animate__backInLeft'>GifExpert-App</h2>
    <AddCategory setCategories={setCategories}/>
    <hr />

    <ol className='mt-2'>
      { 
        categories.map( category => (
              <GifGrid 
              key={category}
              category={category}
              />
        ))
      }
    </ol>

    </>
  )

}
export default GifExpertApp;