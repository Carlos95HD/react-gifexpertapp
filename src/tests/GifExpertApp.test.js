import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp'

describe('Test <GifExpertApp />', () => {

  test('Se debe mostrar correctamente GifExpertApp', () => {

    const categories = [ 'Gato', 'Oso' ]

    const wrapper = shallow(<GifExpertApp defaultCategories = { categories } />)
    
    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    
  })

})