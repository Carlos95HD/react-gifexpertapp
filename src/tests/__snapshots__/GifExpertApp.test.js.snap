// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> Se debe mostrar correctamente GifExpertApp 1`] = `
<Fragment>
  <h2
    className="text-4xl font-mono text-gray-800 animate__animated animate__backInLeft"
  >
    GifExpert-App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol
    className="mt-2"
  >
    <GifGrid
      category="Gato"
      key="Gato"
    />
    <GifGrid
      category="Oso"
      key="Oso"
    />
  </ol>
</Fragment>
`;
